pkg/cluster/kubernetes/resource/load.go

```diff
@@ -39,7 +41,16 @@ func Load(base string, paths []string) (map[string]KubeManifest, error) {
 			}
 
 			if !info.IsDir() && filepath.Ext(path) == ".yaml" || filepath.Ext(path) == ".yml" {
-				bytes, err := ioutil.ReadFile(path)
+				var bytes []byte
+				decryptFile, err := shouldDecrypt(path, sopsPattern)
```

Comment:
Not familiar with this codebase so I may be misguided in mentioning this, but the way we usually do this at Mozilla is something like:
```golang
// Try to decrypt the conf using sops or load it as plaintext.
// If the configuration is not encrypted with sops, the error
// sops.MetadataNotFound will be returned, in which case we
// ignore it and continue loading the conf.
confData, err = decrypt.Data(data, "yaml")
if err != nil {
    if err == sops.MetadataNotFound {
        // not an encrypted file
        confData = data
    } else {
        return errors.Wrap(err, "failed to load sops encrypted configuration")
    }
}
```

examples: https://github.com/mozilla-services/foxsec-pipeline-contrib/blob/master/common/configuration.go#L71-L79 and https://github.com/mozilla-services/autograph/blob/master/main.go#L255-L268

Reply:
This has the advantage of not requiring a special filename, which is great!

However, we should confirm that the metadata check is really cheap, otherwise we cannot afford to do it with every file.